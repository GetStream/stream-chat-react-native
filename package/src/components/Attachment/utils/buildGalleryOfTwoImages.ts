import type { Attachment } from 'stream-chat';

import { buildThumbnailGrid } from './buildThumbnailGrid';
import type { GallerySizeAndThumbnailGrid, GallerySizeConfig } from './types';

import type { DefaultAttachmentType } from '../../../types/types';

export function buildGalleryOfTwoImages<At extends DefaultAttachmentType>({
  images,
  sizeConfig,
}: {
  images: Attachment<At>[];
  sizeConfig: GallerySizeConfig;
}): GallerySizeAndThumbnailGrid {
  const { defaultHeight, defaultWidth } = sizeConfig;

  const image1Height = images[0].height || defaultHeight;
  const image1Width = images[0].width || defaultWidth;
  const image2Height = images[1].height || defaultHeight;
  const image2Width = images[1].width || defaultWidth;

  const aspectRatio1 = image1Width / image1Height;
  const aspectRatio2 = image2Width / image2Height;

  // check if one image is landscape and other is portrait or vice versa
  const isLandscape1 = aspectRatio1 > 1;
  const isLandscape2 = aspectRatio2 > 1;

  // Both the images are landscape
  if (isLandscape1 && isLandscape2) {
    return buildThumbnailGrid({
      grid: [[1], [1]],
      images,
      invertedDirections: true,
      sizeConfig,
    });
  }

  if (!isLandscape1 && !isLandscape2) {
    return buildThumbnailGrid({
      grid: [[1], [1]],
      images,
      invertedDirections: false,
      sizeConfig,
    });
  }

  return buildThumbnailGrid({
    grid: [[2, 1]],
    images: isLandscape1 ? images : images.reverse(),
    invertedDirections: true,
    sizeConfig,
  });
}
