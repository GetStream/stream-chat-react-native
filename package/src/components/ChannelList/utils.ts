import type { Channel, StreamChat } from 'stream-chat';

import type { DefaultStreamChatGenerics } from '../../types/types';

type MoveParameters<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = {
  channels: Channel<StreamChatGenerics>[];
  cid: string;
};

export const moveChannelUp = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>({
  channels = [],
  cid,
}: MoveParameters<StreamChatGenerics>) => {
  // get channel from channels
  const index = channels.findIndex((c) => c.cid === cid);
  if (index <= 0) return channels;
  const channel = channels[index];

  // remove channel from current position and add to start
  channels.splice(index, 1);
  channels.unshift(channel);

  return [...channels];
};

type GetParameters<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
> = {
  client: StreamChat<StreamChatGenerics>;
  id: string;
  type: string;
};

export const getChannel = async <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>({
  client,
  id,
  type,
}: GetParameters<StreamChatGenerics>) => {
  const channel = client.channel(type, id);
  await channel.watch();
  return channel;
};

export const DEFAULT_QUERY_CHANNELS_LIMIT = 10;
export const MAX_QUERY_CHANNELS_LIMIT = 30;
