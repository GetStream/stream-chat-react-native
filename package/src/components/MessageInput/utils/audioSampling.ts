export const resampleWaveformData = (waveformData: number[], amplitudesCount: number) =>
  waveformData.length === amplitudesCount
    ? waveformData
    : waveformData.length > amplitudesCount
    ? downSample(waveformData, amplitudesCount)
    : upSample(waveformData, amplitudesCount);

/**
 * The downSample function uses the Largest-Triangle-Three-Buckets (LTTB) algorithm.
 * See the thesis Downsampling Time Series for Visual Representation by Sveinn Steinarsson for more (https://skemman.is/bitstream/1946/15343/3/SS_MSthesis.pdf)
 * @param data
 * @param targetOutputSize
 */
export function downSample(data: number[], targetOutputSize: number): number[] {
  if (data.length <= targetOutputSize || targetOutputSize === 0) {
    return data;
  }

  if (targetOutputSize === 1) return [mean(data)];

  const result: number[] = [];
  // bucket size adjusted due to the fact that the first and the last item in the original data array is kept in target output
  const bucketSize = (data.length - 2) / (targetOutputSize - 2);
  let lastSelectedPointIndex = 0;
  result.push(data[lastSelectedPointIndex]); // Always add the first point
  let maxAreaPoint, maxArea, triangleArea;

  for (let bucketIndex = 1; bucketIndex < targetOutputSize - 1; bucketIndex++) {
    const previousBucketRefPoint = data[lastSelectedPointIndex];
    const nextBucketMean = getNextBucketMean(data, bucketIndex, bucketSize);

    const currentBucketStartIndex = Math.floor((bucketIndex - 1) * bucketSize) + 1;
    const nextBucketStartIndex = Math.floor(bucketIndex * bucketSize) + 1;
    const countUnitsBetweenAtoC = 1 + nextBucketStartIndex - currentBucketStartIndex;

    maxArea = triangleArea = -1;

    for (
      let currentPointIndex = currentBucketStartIndex;
      currentPointIndex < nextBucketStartIndex;
      currentPointIndex++
    ) {
      const countUnitsBetweenAtoB = Math.abs(currentPointIndex - currentBucketStartIndex) + 1;
      const countUnitsBetweenBtoC = countUnitsBetweenAtoC - countUnitsBetweenAtoB;
      const currentPointValue = data[currentPointIndex];

      triangleArea = triangleAreaHeron(
        triangleBase(Math.abs(previousBucketRefPoint - currentPointValue), countUnitsBetweenAtoB),
        triangleBase(Math.abs(currentPointValue - nextBucketMean), countUnitsBetweenBtoC),
        triangleBase(Math.abs(previousBucketRefPoint - nextBucketMean), countUnitsBetweenAtoC),
      );

      if (triangleArea > maxArea) {
        maxArea = triangleArea;
        maxAreaPoint = data[currentPointIndex];
        lastSelectedPointIndex = currentPointIndex;
      }
    }

    if (typeof maxAreaPoint !== 'undefined') result.push(maxAreaPoint);
  }

  result.push(data[data.length - 1]); // Always add the last point

  return result;
}

const triangleAreaHeron = (a: number, b: number, c: number) => {
  const s = (a + b + c) / 2;
  return Math.sqrt(s * (s - a) * (s - b) * (s - c));
};

const triangleBase = (a: number, b: number) => Math.sqrt(Math.pow(a, 2) + Math.pow(b, 2));
const mean = (values: number[]) => values.reduce((acc, value) => acc + value, 0) / values.length;
export const divMod = (num: number, divisor: number) => [Math.floor(num / divisor), num % divisor];

const getNextBucketMean = (data: number[], currentBucketIndex: number, bucketSize: number) => {
  const nextBucketStartIndex = Math.floor(currentBucketIndex * bucketSize) + 1;
  let nextNextBucketStartIndex = Math.floor((currentBucketIndex + 1) * bucketSize) + 1;
  nextNextBucketStartIndex =
    nextNextBucketStartIndex < data.length ? nextNextBucketStartIndex : data.length;

  return mean(data.slice(nextBucketStartIndex, nextNextBucketStartIndex));
};

export const upSample = (values: number[], targetSize: number) => {
  if (!values.length) {
    console.warn('Cannot extend empty array of amplitudes.');
    return values;
  }

  if (values.length > targetSize) {
    console.warn('Requested to extend the waveformData that is longer than the target list size');
    return values;
  }

  if (targetSize === values.length) return values;

  // eslint-disable-next-line prefer-const
  let [bucketSize, remainder] = divMod(targetSize, values.length);
  const result: number[] = [];

  for (let i = 0; i < values.length; i++) {
    const extra = remainder && remainder-- ? 1 : 0;
    result.push(...Array(bucketSize + extra).fill(values[i]));
  }
  return result;
};
