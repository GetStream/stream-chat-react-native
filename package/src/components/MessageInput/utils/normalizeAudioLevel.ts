/**
 * Utility function to normalize the audio level.
 */
export const normalizeAudioLevel = (value: number, lowerBound: number) => {
  // For Native CLI, the lower bound is around -50
  const upperBound = 0;

  const delta = upperBound - lowerBound;

  // In Native CLI Android, the value is undefined for loud audio
  if (value === undefined) return 1;

  if (value < lowerBound) {
    return 0;
  } else if (value >= upperBound) {
    return 1;
  } else {
    return Math.abs((value - lowerBound) / delta);
  }
};
