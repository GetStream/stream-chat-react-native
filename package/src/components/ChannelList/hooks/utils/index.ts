import { Channel, ChannelSortBase } from 'stream-chat';

import { DefaultStreamChatGenerics } from '../../../../types/types';
import { ChannelListProps } from '../../ChannelList';

export const isChannelPinned = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  channel: Channel<StreamChatGenerics>,
) => {
  if (!channel) {
    return false;
  }

  const member = channel.state.membership;

  return !!member?.pinned_at;
};

export const isChannelArchived = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  channel: Channel<StreamChatGenerics>,
) => {
  if (!channel) {
    return false;
  }

  const member = channel.state.membership;

  return !!member?.archived_at;
};

export const shouldConsiderArchivedChannels = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  filters: ChannelListProps<StreamChatGenerics>['filters'],
) => {
  if (!filters) {
    return false;
  }

  return typeof filters.archived === 'boolean';
};

export const extractSortValue = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>({
  atIndex,
  sort,
  targetKey,
}: {
  atIndex: number;
  targetKey: keyof ChannelSortBase<StreamChatGenerics>;
  sort?: ChannelListProps<StreamChatGenerics>['sort'];
}) => {
  if (!sort) {
    return null;
  }
  let option: null | ChannelSortBase<StreamChatGenerics> = null;

  if (Array.isArray(sort)) {
    option = sort[atIndex] ?? null;
  } else {
    let index = 0;
    for (const key in sort) {
      if (index !== atIndex) {
        index++;
        continue;
      }

      if (key !== targetKey) {
        return null;
      }

      option = sort;

      break;
    }
  }

  return option?.[targetKey] ?? null;
};

/**
 * Returns true only if `{ pinned_at: -1 }` or `{ pinned_at: 1 }` option is first within the `sort` array.
 */
export const shouldConsiderPinnedChannels = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>(
  sort: ChannelListProps<StreamChatGenerics>['sort'],
) => {
  const value = extractSortValue({
    atIndex: 0,
    sort,
    targetKey: 'pinned_at',
  });

  if (typeof value !== 'number') {
    return false;
  }

  return Math.abs(value) === 1;
};

export function findPinnedAtSortOrder<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>({ sort }: { sort: ChannelListProps<StreamChatGenerics>['sort'] }) {
  if (!sort) {
    return null;
  }

  if (Array.isArray(sort)) {
    const [option] = sort;

    if (!option?.pinned_at) {
      return null;
    }

    return option.pinned_at;
  } else {
    if (!sort.pinned_at) {
      return null;
    }

    return sort.pinned_at;
  }
}

export function findLastPinnedChannelIndex<
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>({ channels }: { channels: Channel<StreamChatGenerics>[] }) {
  let lastPinnedChannelIndex: number | null = null;

  for (const channel of channels) {
    if (!isChannelPinned(channel)) {
      break;
    }

    if (typeof lastPinnedChannelIndex === 'number') {
      lastPinnedChannelIndex++;
    } else {
      lastPinnedChannelIndex = 0;
    }
  }

  return lastPinnedChannelIndex;
}
