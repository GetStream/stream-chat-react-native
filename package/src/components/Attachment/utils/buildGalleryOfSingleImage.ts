import type { Attachment } from 'stream-chat';

import { buildThumbnail } from './buildThumbnail';
import { getAspectRatio } from './getAspectRatio';
import type { GallerySizeAndThumbnailGrid, GallerySizeConfig } from './types';

import type { DefaultAttachmentType } from '../../../types/types';

function clamp(number: number, min: number, max: number) {
  return Math.min(Math.max(number, min), max);
}

function getContainerSize<At extends DefaultAttachmentType = DefaultAttachmentType>({
  image,
  sizeConfig,
}: {
  image: Attachment<At>;
  sizeConfig: GallerySizeConfig;
}) {
  const { height, width } = image;
  const { gridHeight, gridWidth, maxHeight, maxWidth, minHeight, minWidth } = sizeConfig;

  if (!height || !width) {
    return { height: gridHeight, width: gridWidth };
  }

  const aspectRatio = getAspectRatio(image);

  if (aspectRatio <= 1) {
    const containerHeight = clamp(height, minHeight, maxHeight);
    const containerWidth = clamp(containerHeight * aspectRatio, minWidth, maxWidth);

    return {
      height: containerHeight,
      width: containerWidth,
    };
  }

  const containerWidth = clamp(width, minWidth, maxWidth);
  const containerHeight = clamp(containerWidth / aspectRatio, minHeight, maxHeight);

  return {
    height: containerHeight,
    width: containerWidth,
  };
}

export function buildGalleryOfSingleImage<
  At extends DefaultAttachmentType = DefaultAttachmentType,
>({
  image,
  sizeConfig,
}: {
  image: Attachment<At>;
  sizeConfig: GallerySizeConfig;
}): GallerySizeAndThumbnailGrid {
  const container = getContainerSize({
    image,
    sizeConfig,
  });

  const thumbnail = buildThumbnail({
    image,
    ...container,
  });

  const column = [thumbnail];

  return {
    ...container,
    thumbnailGrid: [column],
  };
}
