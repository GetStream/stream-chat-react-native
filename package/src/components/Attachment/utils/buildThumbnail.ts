import type { ImageResizeMode } from 'react-native';

import type { Attachment } from 'stream-chat';

import type { Thumbnail } from './types';

import type { DefaultAttachmentType } from '../../../types/types';

import { getResizedImageUrl } from '../../../utils/getResizedImageUrl';

export function buildThumbnail<At extends DefaultAttachmentType>({
  height,
  image,
  resizeMode,
  width,
}: {
  height: number;
  image: Attachment<At>;
  width: number;
  resizeMode?: ImageResizeMode;
}): Thumbnail {
  return {
    height,
    resizeMode: resizeMode
      ? resizeMode
      : ((image.height && image.width ? 'contain' : 'cover') as ImageResizeMode),
    url: getResizedImageUrl({
      height,
      image,
      width,
    }),
    width,
  };
}
