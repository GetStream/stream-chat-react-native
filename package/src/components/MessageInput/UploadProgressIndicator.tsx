import React from 'react';
import {
  ActivityIndicator,
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

import { useTheme } from '../../contexts/themeContext/ThemeContext';
import { Refresh } from '../../icons';
import { ProgressIndicatorTypes } from '../../utils/utils';

const REFRESH_ICON_SIZE = 18;

const styles = StyleSheet.create({
  activityIndicatorContainer: {
    alignItems: 'center',
    bottom: 0,
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
  },
  container: {
    alignItems: 'center',
    height: '100%',
    justifyContent: 'center',
    position: 'absolute',
    width: '100%',
  },
  overflowHidden: {
    overflow: 'hidden',
  },
  overlay: {
    alignItems: 'center',
    height: '100%',
    justifyContent: 'center',
    opacity: 0,
    position: 'absolute',
    width: '100%',
  },
  retryButtonContainer: {
    alignItems: 'center',
    bottom: 0,
    justifyContent: 'center',
    left: 0,
    position: 'absolute',
    right: 0,
    top: 0,
  },
});

export type UploadProgressIndicatorProps = {
  /** Action triggered when clicked indicator */
  action?: (event: GestureResponderEvent) => void;
  /** style */
  style?: StyleProp<ViewStyle>;
  /** Type of active indicator */
  type?: 'in_progress' | 'retry' | 'not_supported' | 'inactive' | null;
};

export const UploadProgressIndicator: React.FC<UploadProgressIndicatorProps> = (props) => {
  const { action, children, style, type } = props;

  const {
    theme: {
      colors: { overlay: overlayColor },
      messageInput: {
        uploadProgressIndicator: { container, overlay },
      },
    },
  } = useTheme();

  return type === ProgressIndicatorTypes.INACTIVE ? (
    <View style={[styles.overflowHidden, style]} testID='inactive-upload-progress-indicator'>
      {children}
    </View>
  ) : (
    <View style={[styles.overflowHidden, style]} testID='active-upload-progress-indicator'>
      {children}
      <View style={[styles.overlay, { backgroundColor: overlayColor }, overlay]} />
      <View
        style={[
          type === ProgressIndicatorTypes.NOT_SUPPORTED ? styles.overflowHidden : styles.container,
          { backgroundColor: overlayColor },
          container,
        ]}
        testID='not-supported-upload-progress-indicator'
      >
        {type === ProgressIndicatorTypes.IN_PROGRESS && <InProgressIndicator />}
        {type === ProgressIndicatorTypes.RETRY && <RetryIndicator action={action} />}
      </View>
    </View>
  );
};

const InProgressIndicator = () => {
  const {
    theme: {
      colors: { white_smoke },
    },
  } = useTheme();

  return (
    <View style={styles.activityIndicatorContainer}>
      <ActivityIndicator color={white_smoke} testID='upload-progress-indicator' />
    </View>
  );
};

const RetryIndicator = ({ action }: Pick<UploadProgressIndicatorProps, 'action'>) => {
  const {
    theme: {
      colors: { white_smoke },
    },
  } = useTheme();

  return (
    <TouchableOpacity onPress={action} style={styles.retryButtonContainer}>
      <Refresh
        height={REFRESH_ICON_SIZE}
        pathFill={white_smoke}
        testID='retry-upload-progress-indicator'
        width={REFRESH_ICON_SIZE}
      />
    </TouchableOpacity>
  );
};

UploadProgressIndicator.displayName =
  'UploadProgressIndicator{messageInput{uploadProgressIndicator}}';
