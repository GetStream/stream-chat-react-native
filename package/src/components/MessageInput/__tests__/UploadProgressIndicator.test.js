import React from 'react';

import { fireEvent, render, waitFor } from '@testing-library/react-native';

import { ThemeProvider } from '../../../contexts/themeContext/ThemeContext';
import { ProgressIndicatorTypes } from '../../../utils/utils';
import { UploadProgressIndicator } from '../UploadProgressIndicator';

describe('UploadProgressIndicator', () => {
  it('should render an inactive UploadProgressIndicator', async () => {
    const action = jest.fn();

    const { queryByTestId, toJSON } = render(
      <ThemeProvider>
        <UploadProgressIndicator
          action={action}
          type={ProgressIndicatorTypes.INACTIVE}
        ></UploadProgressIndicator>
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(queryByTestId('active-upload-progress-indicator')).toBeFalsy();
      expect(queryByTestId('inactive-upload-progress-indicator')).toBeTruthy();
      expect(action).toHaveBeenCalledTimes(0);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render an active UploadProgressIndicator', async () => {
    const action = jest.fn();

    const { queryByTestId, toJSON } = render(
      <ThemeProvider>
        <UploadProgressIndicator
          action={action}
          type={ProgressIndicatorTypes.IN_PROGRESS}
        ></UploadProgressIndicator>
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(queryByTestId('active-upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('inactive-upload-progress-indicator')).toBeFalsy();
      expect(action).toHaveBeenCalledTimes(0);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render an active UploadProgressIndicator and not-supported indicator', async () => {
    const action = jest.fn();

    const { queryByTestId, toJSON } = render(
      <ThemeProvider>
        <UploadProgressIndicator
          action={action}
          type={ProgressIndicatorTypes.NOT_SUPPORTED}
        ></UploadProgressIndicator>
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(queryByTestId('active-upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('not-supported-upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('inactive-upload-progress-indicator')).toBeFalsy();
      expect(action).toHaveBeenCalledTimes(0);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render an active UploadProgressIndicator and in-progress indicator', async () => {
    const action = jest.fn();

    const { queryByTestId, toJSON } = render(
      <ThemeProvider>
        <UploadProgressIndicator
          action={action}
          type={ProgressIndicatorTypes.IN_PROGRESS}
        ></UploadProgressIndicator>
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(queryByTestId('active-upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('inactive-upload-progress-indicator')).toBeFalsy();
      expect(action).toHaveBeenCalledTimes(0);
    });

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  it('should render an active UploadProgressIndicator and retry indicator', async () => {
    const action = jest.fn();

    const { getByTestId, queryByTestId, toJSON } = render(
      <ThemeProvider>
        <UploadProgressIndicator
          action={action}
          type={ProgressIndicatorTypes.RETRY}
        ></UploadProgressIndicator>
      </ThemeProvider>,
    );

    await waitFor(() => {
      expect(queryByTestId('active-upload-progress-indicator')).toBeTruthy();
      expect(queryByTestId('upload-progress-indicator')).toBeFalsy();
      expect(queryByTestId('inactive-upload-progress-indicator')).toBeFalsy();
      expect(queryByTestId('retry-upload-progress-indicator')).toBeTruthy();
      expect(action).toHaveBeenCalledTimes(0);
    });

    fireEvent.press(getByTestId('retry-upload-progress-indicator'));

    await waitFor(() => expect(action).toHaveBeenCalledTimes(1));

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
