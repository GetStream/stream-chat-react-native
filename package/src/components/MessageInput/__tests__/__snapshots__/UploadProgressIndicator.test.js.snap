// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProgressIndicator should render an active UploadProgressIndicator 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
  testID="active-upload-progress-indicator"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#000000CC",
        },
        Object {},
      ]
    }
    testID="not-supported-upload-progress-indicator"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        color="#F2F2F2"
        testID="upload-progress-indicator"
      />
    </View>
  </View>
</View>
`;

exports[`UploadProgressIndicator should render an active UploadProgressIndicator and in-progress indicator 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
  testID="active-upload-progress-indicator"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#000000CC",
        },
        Object {},
      ]
    }
    testID="not-supported-upload-progress-indicator"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        color="#F2F2F2"
        testID="upload-progress-indicator"
      />
    </View>
  </View>
</View>
`;

exports[`UploadProgressIndicator should render an active UploadProgressIndicator and not-supported indicator 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
  testID="active-upload-progress-indicator"
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#000000CC",
        },
        Object {},
      ]
    }
    testID="not-supported-upload-progress-indicator"
  />
</View>
`;

exports[`UploadProgressIndicator should render an active UploadProgressIndicator and retry indicator 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
  testID="active-upload-progress-indicator"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#000000CC",
        },
        Object {},
      ]
    }
    testID="not-supported-upload-progress-indicator"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={18}
        bbWidth={18}
        focusable={false}
        height={18}
        meetOrSlice={0}
        minX={0}
        minY={0}
        pathFill="#F2F2F2"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        testID="retry-upload-progress-indicator"
        vbHeight={24}
        vbWidth={24}
        width={18}
      >
        <RNSVGGroup>
          <RNSVGPath
            clipRule={0}
            d="M12.868 4.221a8 8 0 102.834 14.251 1 1 0 00-1.175-1.618A6 6 0 1112.4 6.166c1.31.314 2.52.803 3.31 2.12.154.254.266.485.348.708h-1.152a1 1 0 00-.81 1.585l2.375 3.291a1 1 0 001.621 0l2.376-3.291a1 1 0 00-.811-1.585h-1.509c-.128-.639-.377-1.24-.867-1.947a8 8 0 00-4.414-2.826z"
            fill={4294111986}
            fillRule={0}
            propList={
              Array [
                "fill",
                "fillRule",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`UploadProgressIndicator should render an inactive UploadProgressIndicator 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      undefined,
    ]
  }
  testID="inactive-upload-progress-indicator"
/>
`;
